tests/Support/Constraint/CssConstraint.php

```diff
@@ -50,20 +49,28 @@ protected static function getCssNeedleRegularExpressionPattern(string $needle):
     {
         $needleMatcher = \preg_replace_callback(
             self::CSS_REGEX_PATTERN,
-            static function (array $matches): string {
-                switch (true) {
-                    case isset($matches[1]) && $matches[1] !== '':
-                        return '\\s*+' . \preg_quote($matches[1], '/') . '\\s*+';
-                    case isset($matches[2]) && $matches[2] !== '':
-                        return '\\s*+';
-                    case isset($matches[3]) && $matches[3] !== '':
-                        return \preg_quote($matches[3], '/') . '\\s*+';
-                    default:
-                        return \preg_quote($matches[0], '/');
-                }
-            },
+            [self::class, 'getCssNeedleRegularExpressionReplacement'],
             $needle
         );
         return '/' . $needleMatcher . '/';
     }
+
+    /**
+     * @param string[] $matches array of matches for {@see CSS_REGEX_PATTERN}
+     *
+     * @return string replacement string, which may be `$matches[0]` if no alteration is needed
+     */
+    private static function getCssNeedleRegularExpressionReplacement(array $matches): string
+    {
+        switch (true) {
+            case isset($matches[1]) && $matches[1] !== '':
+                return '\\s*+' . \preg_quote($matches[1], '/') . '\\s*+';
+            case isset($matches[2]) && $matches[2] !== '':
+                return '\\s*+';
+            case isset($matches[3]) && $matches[3] !== '':
+                return \preg_quote($matches[3], '/') . '\\s*+';
+            default:
+                return \preg_quote($matches[0], '/');
```

Comment:
We still have the early returns here. Please let's change this to put the result in a local variable in the different cases and return that local variable at the end of the function.

Reply:
Hmm.  It's arguable that they are not early returns because there is no more code on any execution branch.  But I will use a temporary local variable...